### DISCUSSION TITLE
[ Data - Core ] -  💡Idea - Add "icon" and "color" FieldTypes / columntypes

### DISCUSSION THREAD

**Discussion by poirazis** [Budibase/budibase#13042] [Ideas]
In the modern world , colors and icons are extensively used in data visualization.
The proposal is that two new FieldTypes are provided ( essentially string subtypes )
an Icon column ( which will hold a string of the remix icon class name ) and a Color column ( string again )

**Comment by linear** [comment]
<p><a href="https://linear.app/budibase/issue/BUDI-8016/[-data-core-]-💡idea-add-icon-and-color-fieldtypes-columntypes">BUDI-8016 [ Data - Core ] - 💡Idea - Add "icon" and "color" FieldTypes / columntypes</a></p>

**Comment by melohagan** [comment]
~~Could definitely build custom components for these!~~

**Reply by poirazis** [reply]
The proposal is for these data types to be supported in the back end so they are part of the schema

**Reply by melohagan** [reply]
I understand now - that is a cool idea!

**Reply by poirazis** [reply]
if users can define these as part of their data, to icon / color code their data, the front-end generators will be able to generate much richer screens.

**Comment by andyburgessmd** [comment]
Could they have some kind of conditional or formula part to it too? So if a value in the form is x then the colour would be set to 'blue'?

**Reply by poirazis** [reply]
You can do that now. Use Formula columns to generate the required color, and then in the Front-End controls you would set the Color property of some Text lets say to that Formula column.

You can also use Conditional Custom CSS
